pkg/controller/mizar/mizar-arktos-network-controller.go

```diff
@@ -149,60 +157,150 @@ func (c *MizarArktosNetworkController) syncNetwork(eventKeyWithType KeyWithEvent
 	if err != nil {
+		klog.Errorf("Create VPC and Subnet failed for tenant %v. Error: (%v).", network.Tenant, err)
 		return err
 	}
-	return nil
-}
-
-func (c *MizarArktosNetworkController) processNetworkCreation(network *v1.Network, eventKeyWithType KeyWithEventType) error {
-	//skip update or create if type is not mizar or network status is ready
-	key := eventKeyWithType.Key
-	if network.Spec.Type != mizarNetworkType || network.Status.Phase == v1.NetworkReady {
-		c.recorder.Eventf(network, corev1.EventTypeNormal, "processNetworkCreation", "Type is not mizar, nothing to be done in mizar cluster: %v.", network)
-		return nil
-	}
 
 	msg := &BuiltinsArktosMessage{
 		Name: network.Name,
 		Vpc:  network.Spec.VPCID,
 	}
 
 	response := c.grpcAdaptor.CreateArktosNetwork(c.grpcHost, msg)
-
-	code := response.Code
-	context := response.Message
-
-	switch code {
+	switch response.Code {
 	case CodeType_OK:
-		klog.Infof("Mizar handled arktos network and vpc id update successfully: %s", key)
+		klog.Infof("Mizar handled arktos network %v successfully", key)
 	case CodeType_TEMP_ERROR:
 		klog.Warningf("Mizar hit temporary error for arktos network and vpc id update: %s", key)
-		c.queue.AddRateLimited(eventKeyWithType)
 		return errors.New("Arktos network and vpc id update failed on mizar side, will try again.....")
 	case CodeType_PERM_ERROR:
 		klog.Errorf("Mizar hit permanent error for Arktos network creation for Arktos network: %s", key)
 		return errors.New("Arktos network and vpc id update failed permanently on mizar side")
 	}
 
-	c.recorder.Eventf(network, corev1.EventTypeNormal, "processNetworkCreation", "successfully created network from mizar cluster: %v.", context)
+	c.recorder.Eventf(network, corev1.EventTypeNormal, "SucessfulCreate", "Created Mizar VPC %s and subnet %s for tenant %v", vpc, subnet, network.Tenant)
+
+	klog.V(3).Infof("Created VPC (%s) and Subnet(%s) for tenant %s successfully", vpc, subnet, network.Tenant)
+
+	return nil
+}
+
+func createVpcAndSubnet(vpc, subnet string, dynamicClient dynamic.Interface) error {
+	// Create VPC object
+	vpcIpStart, vpcSpec := generateVPCSpec(vpc)
+	vpcData, err := json.Marshal(vpcSpec)
+	if err != nil {
+		klog.Errorf("Error marshalling VPC %s spec. Error: %v", vpc, err)
+		return err
+	}
+	err = controller.CreateUnstructuredObject(vpcData, dynamicClient)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		klog.Errorf("Error creating VPC %s. Error: %v", vpc, err)
+		return err
+	}
+
+	// Create Subnet object
+	subnetSpec, err := generateSubnetSpec(vpc, subnet, vpcIpStart)
+	if err != nil {
+		klog.Errorf("Error getting Subnet %s spec. Error: %v", subnet, err)
+		return err
+	}
+	err = controller.CreateUnstructuredObject(subnetSpec, dynamicClient)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		klog.Errorf("Error creating Subnet %s for VPC %s. Error: %v", subnetSpec, vpc, err)
+		return err
+	}
+
 	return nil
 }
+
+func generateVPCSpec(vpcName string) (int, *MizarVPC) {
```

Comment:
suggest to comment the function for the current limits in case future code review ask the same. for now, user specified vpc cidr is not supported, hence the impl. below.

Reply:
Done